ports/esp32/main.c

```diff
@@ -237,7 +238,10 @@ void *esp_native_code_commit(void *buf, size_t len, void *reloc) {
     len = (len + 3) & ~3;
     size_t len_node = sizeof(native_code_node_t) + len;
     native_code_node_t *node = heap_caps_malloc(len_node, MALLOC_CAP_EXEC);
-    if (node == NULL) {
+    // Executable check is workaround for ESP-IDF bug where RTC FAST memory
+    // incorrectly returned here
```

Comment:
Should we just keep this "workaround" here ~forever, as a fail safe in case the IDF has similar issues in the future?  If so, maybe tweak the comment "Executable check is workaround for ESP-IDF bug where non-executable memory (eg RTC fast memory) is incorrectly returned here.".

Or, guard it with a ESP32S2-only config?

Reply:
Since you reviewed I updated it with a link to the upstream issue, but it seems fair enough to keep it in.

I actually noticed a similar problem on ESP32 when testing this (mentioned in the upstream bug report), but in this case I'm not sure if the problem is in the `heap_caps_malloc()` result or the `esp_ptr_executable()` result...

Reply:
Yeah it seems like `esp_ptr_executable()` returns false on original ESP32 for some addresses which are executable!! So I'll narrow this down to just the ESP32-S2 target.

Reply:
Have gated to ESP32-S2 only, to avoid unnecessary failures on ESP32...

Reply:
Oh, wow, bugs all round.

What's here now is good.